lib/src/download_utils.dart

```diff
@@ -114,18 +114,42 @@ enum UrlStatus {
   exists,
 }
 
+/// The method returns the cached existence status of the [url]
+/// - returns `true` if the [url] exists,
+/// - returns `false` if the [url] is missing or has issues serving,
+/// - returns `null` if the [url] status is not in the cache.
+typedef GetCachedUrlStatus = Future<bool?> Function(String url);
```

Comment:
If these always come as a pair, perhaps it would be better to have an interface with two methods?